pymeasure/instruments/keithley/keithley2510.py

```diff
@@ -0,0 +1,207 @@
+        ":SOURce:TEMPerature:PROTection:STATe?",
+        ":SOURce:TEMPerature:PROTection:STATe %d",
+        """Control whether temperature protection is enabled. Takes values True or False.
+        The convenience methods :meth:`~.Keithley2510.enable_temperature_protection`
+        and :meth:`~.Keithley2510.disable_temperature_protection` can also be used.""",
+        validator=strict_discrete_set,
+        values={True: 1, False: 0},
+        map_values=True,
+    )
+
+    temperature_protection_low = Instrument.control(
+        ":SOURce:TEMPerature:PROTection:LOW?",
+        ":SOURce:TEMPerature:PROTection:LOW %g",
+        """Control the lower temperature limit in degrees centigrade.
+        Takes values from -50 to 250.""",
+        validator=truncated_range,
+        values=[-50, 250],
+    )
+
+    temperature_protection_high = Instrument.control(
+        ":SOURce:TEMPerature:PROTection:HIGH?",
+        ":SOURce:TEMPerature:PROTection:HIGH %g",
+        """Control the upper temperature limit in degrees centigrade.
+        Takes values from -50 to 250.""",
+        validator=truncated_range,
+        values=[-50, 250],
+    )
+
+    @property
+    def temperature_protection_range(self):
+        """Control the lower and upper temperature limits in degrees centigrade.
+        Two-tuple of floats (lower_limit, upper_limit) which can take values from -50 to 250."""
+        return (self.temperature_protection_low, self.temperature_protection_high)
+
+    @temperature_protection_range.setter
+    def temperature_protection_range(self, temp_range):
+        self.temperature_protection_low = temp_range[0]
+        self.temperature_protection_high = temp_range[1]
+
+    ###########
+    # Methods #
+    ###########
+
+    def enable_source(self):
+        """Enables the source."""
+        self.write("OUTPUT ON")
+
+    def disable_source(self):
+        """Disables the source."""
+        self.write("OUTPUT OFF")
+
+    def enable_temperature_protection(self):
+        """Enable temperature protection."""
+        self.write(":SOURce:TEMPerature:PROTection:STATe ON")
+
+    def disable_temperature_protection(self):
+        """Disable temperature protection."""
+        self.write(":SOURce:TEMPerature:PROTection:STATe OFF")
+
+    def is_temperature_stable(self, tolerance=0.1, period=10, points=64):
+        """Determine whether the temperature is stable at the temperature setpoint over a specified
+        period.
+
+        :param tolerance: Maximum allowed deviation from temperature setpoint,
+            in degrees Centigrade.
+        :param period: Time period over which stability is checked, in seconds.
+        :param points: Number of points to collect within the period.
+        :return: True if stable, False otherwise.
+        """
+
+        delay = period / points
+
+        temp_array = []
+
+        for i in range(points):
+            temp_array.append(self.temperature)
+            sleep(delay)
```

Comment:
another idea:
you compare each temperature measurement with the setpoint. If any value exceeds the temperature range, you return False.